src/components/Congress/Motions/index.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import Card from '../../Card';
+import styles from './styles.module.scss';
+import Button from '../../Button/Button';
+
+// REDUX
+import { congressActions } from '../../../redux/actions';
+import {
+  congressSelectors,
+  blockchainSelectors,
+} from '../../../redux/selectors';
+
+export default function Motions() {
+  const dispatch = useDispatch();
+  const motions = useSelector(congressSelectors.congressMotions);
+
+  useEffect(() => {
+    dispatch(congressActions.getMotions.call());
+  }, [dispatch]);
+
+  return (
+    <div>
+      {motions.map(({ proposal, proposalOf, voting }) => {
+        const readableProposal = proposal.toHuman();
+        const readableProposalOf = proposalOf.toHuman();
+        const readableVoting = voting.toHuman();
+
+        return (
+          <Motion
+            key={readableProposal}
+            proposal={readableProposal}
+            proposalOf={readableProposalOf}
+            voting={readableVoting}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+function Motion({ proposal, proposalOf, voting }) {
+  const dispatch = useDispatch();
+  const sender = useSelector(blockchainSelectors.userWalletAddressSelector);
+
+  return (
+    <Card
+      key={proposal}
+      title={`${proposalOf.method}`}
+      className={styles.cardProposalsSection}
+    >
+      <div>
+        <div className={styles.metaInfoLine}>
+          <p>
+            Proposal id:
+            {proposal}
+          </p>
+          <p>
+            Aye
+            <b>
+              {voting.ayes.length}
+              /
+              {voting.threshold}
+            </b>
+          </p>
+        </div>
+        <p>
+          Threshold:
+          {voting.threshold}
+        </p>
+
+        <pre>{JSON.stringify(proposalOf.args, null, 2)}</pre>
+        <div className={styles.buttonsContainer}>
+          {!voting.ayes.includes(sender) && (
+            <Button
+              medium
+              primary
+              onClick={() => dispatch(
+                congressActions.voteAtMotions.call({
+                  readableProposal: proposal,
+                  index: voting.index,
+                  vote: true,
+                }),
+              )}
+            >
+              Vote aye
+            </Button>
+          )}
+          {!voting.nays.includes(sender) && (
```

Comment:
Hmm... does the blockchain allow us to vote nay if we've already voted aye (and vice versa)? Does it change the vote, fail or is ignored?

Reply:
We can not vote the same (it is handled in code by throwing error). But when you are trying to vote opposite to your already existing vote. So when we voted "nay" we can not vote "nay" but we can vote "aye"